holoviews/util/__init__.py

```diff
@@ -77,44 +77,74 @@ class opts(param.ParameterizedFunction):
        to strict (default), any invalid keywords are simply skipped. If
        strict, invalid keywords prevent the options being applied.""")
 
-    def __call__(self, *args, **params):
+    _deprecate_magics_call = 'error' # 'warn', 'error' or None
 
-        if args and set(params.keys()) - set(['strict']):
-            raise TypeError('When used with positional arguments, hv.opts accepts only strings and dictionaries, not keywords.')
+    def __call__(self, *args, **params):
         if params and not args:
             return Options(**params)
 
-        p = param.ParamOverrides(self, params)
-        if len(args) not in [1,2]:
-            raise TypeError('The opts utility accepts one or two positional arguments.')
-        elif len(args) == 1:
-            options, obj = args[0], None
+        if len(args) == 1:
+            msg = ("Use opts.linemagic instead as a direct replacement "
+                   "although opts.defaults is the recommended approach.")
+
+            if self._deprecate_magics_call == 'error':
+                raise DeprecationWarning(msg)
```

Comment:
I suppose I should probably use ``DeprecationWarning`` or ``PendingDeprecationWarning`` in my deprecation PR as well.